pkg/kubelet/cm/dra/manager.go

```diff
@@ -28,31 +28,41 @@ import (
 	"k8s.io/dynamic-resource-allocation/resourceclaim"
 	"k8s.io/klog/v2"
 	dra "k8s.io/kubernetes/pkg/kubelet/cm/dra/plugin"
+	"k8s.io/kubernetes/pkg/kubelet/cm/dra/state"
```

Comment:
podresources API does not require state management in addition to what the resource managers would require anyway. In other words: how does DRA recover on kubelet restart?

*IF* DRA requires/benefits from this state management, I'd spin off to its own PR.

Reply:
make sense, I will fill a bug that with lose the DRA cache on kubelet restart and this cause UnprepareResources  not to be called and resources not to be clean. 
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/dra/manager.go#L196. 
and all the checkpointing mechanism will be handle in that PR. 
And here I will keep only the pod resource api kep changes.

Reply:
The issue for DRA recover  kubelet restart https://github.com/kubernetes/kubernetes/issues/115902

Reply:
This PR is depended on this PR https://github.com/kubernetes/kubernetes/pull/115912